core/resources/material.glsl

```diff
@@ -0,0 +1,21 @@
+struct Material {
+	#ifdef MATERIAL_EMISSION
+	vec4 emission;
+	#endif
+
+	#ifdef MATERIAL_AMBIENT
+	vec4 ambient;
+	#endif 
+
+	#ifdef MATERIAL_DIFFUSE
+	vec4 diffuse;
+	#endif
+
+	#ifdef MATERIAL_SPECULAR
+	vec4 specular;
+	float shininess;
+	#endif
+};
+
+uniform Material u_material;
+Material g_material = u_material;
```

Comment:
Is `g_material` instantiated as a copy of `u_material` because we want to allow the shader to modify material properties ad hoc? Just clarifying (could add a comment to this effect).